File: src/main/java/com/pauljoda/modularsystems/storage/tiles/TileStorageCrafting.java
package com.pauljoda.modularsystems.storage.tiles;

/**
 * Modular-Systems
 * Created by Paul Davis on 7/26/2015
 */
public class TileStorageCrafting extends TileStorageBasic {
    @Override
    public void addedToNetwork() {
        super.addedToNetwork();
        if(getCore() != null) {
            getCore().setHasCraftingUpgrade(true);
            worldObj.markBlockForUpdate(core.x, core.y, core.z);

        }
    }

    @Override
    public void removedFromNetwork() {
        super.removedFromNetwork();
        if(getCore() != null) {
            getCore().setHasCraftingUpgrade(false);
            worldObj.markBlockForUpdate(core.x, core.y, core.z);
        }
    }
}

File: src/main/scala/com/teambr/modularsystems/storage/tiles/TileStorageCrafting.scala
package com.teambr.modularsystems.storage.tiles

/**
 * This file was created for Modular-Systems
 *
 * Modular-Systems is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis <pauljoda>
 * @since August 08, 2015
 */
class TileStorageCrafting extends TileEntityStorageExpansion {
    /**
     * Called after this has been added to a network
     */
    override def addedToNetwork() : Unit = {
        getCore match {
            case Some(theCore) =>
                theCore.hasCraftingUpgrade = true
                worldObj.markBlockForUpdate(theCore.getPos)
            case _ =>
        }
    }

    /**
     * Called right before this is removed from a network
     */
    override def removedFromNetwork() : Unit = {
        getCore match {
            case Some(theCore) =>
                theCore.hasCraftingUpgrade = false
                worldObj.markBlockForUpdate(theCore.getPos)
            case _ =>
        }
    }
}
